commit 9c48f201c099c66fb8754e9e495b0d459c3bff55
Author: Robert Rozic <rooobertinhooo@gmail.com>
Date:   Fri Jan 24 15:21:33 2014 +0100

    Fix shutdown/reboot bug

diff --git a/libcutils/android_reboot.c b/libcutils/android_reboot.c
index 326d402..ab836ec 100644
--- a/libcutils/android_reboot.c
+++ b/libcutils/android_reboot.c
@@ -93,7 +93,7 @@ static void remount_ro(void)
 
 
     /* Now poll /proc/mounts till it's done */
-    while (!remount_ro_done() && (cnt < 3600)) {
+    while (!remount_ro_done() && (cnt < 50)) {
         usleep(100000);
         cnt++;
     }
diff --git a/libutils/Flattenable.cpp b/libutils/Flattenable.cpp
index 39135e3..1f2ffaa 100644
--- a/libutils/Flattenable.cpp
+++ b/libutils/Flattenable.cpp
@@ -21,4 +21,4 @@ namespace android {
 Flattenable::~Flattenable() {
 }
 
-}; // namespace android
\ No newline at end of file
+}; // namespace android
